src/Providers/VisualStudioService.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace clio.Providers
+{
+	public class VisualStudioService
+	{
+		string accessToken;
+
+		public VisualStudioService ()
+		{
+		}
+
+		public void Login (string pat)
+		{
+			accessToken = pat;
+		}
+
+		static string EncodeForUrl (string s)
+		{
+			return s.Replace (" ", "%20");
+		}
+
+		public async Task<VisualStudioBug> GetBug (string account, int bugId)
+		{
+			try
+			{
+				try
+				{
+					var httpService = new HttpService ();
+
+					var url = string.Format ("https://{0}.visualstudio.com/_apis/wit/workitems/{1}?$expand=all&api-version=2.2", EncodeForUrl (account), bugId);
+					var res = await httpService.PostAsync (url, user: "", pwd: accessToken, contentType: "application/json-patch+json", method: "GET");
+					return (VisualStudioBug)JsonConvert.DeserializeObject (res, typeof (VisualStudioBug));
+				}
+				catch (WebException ex)
+				{
+					ParseError (ex);
+					throw;
+				}
+			}
+			catch
+			{
+				// TODO: log errors somehow
```

Comment:
Should this scream in explain? Or is it safe to ignore.

Reply:
yeah, forgot about Explain for that. I just didn't want to crash the entire run for weird / network issues.